### ISSUE TITLE
Suggestion: Add an option to use a personal VPN

### ISSUE THREAD

**Issue by ShapeShifter499** [lkorth/auto-fi#2]
I guess I understand this is your VPN.   If this is based off of the ics-openvpn Android app, could there be an option to use a different VPN configuration?

**Comment by ShapeShifter499** [comment]
I'm not sure if using another VPN app on top of this app will mess up the auto-login feature.  I've had times when a VPN app was trying to connect while I wanted to log in on a open wifi and that caused issues.

**Comment by lkorth** [comment]
Yes, I currently run a server that acts as the VPN for this app. It wouldn't be too difficult to add support for a custom OpenVPN configuration so you could use your own server. If you're interested in working on this, please feel free to submit a PR.

It depends on the VPN, Auto Fi binds to the WiFi network and attempts to log in when needed, this usually happens pretty quickly and if the VPN cannot connect because of a captive portal traffic shouldn't be routed through the VPN yet and Auto Fi should be able to bypass the captive portal.

**Comment by ReekyMarko** [comment]
Just a "me too" comment. This would be very useful

**Comment by FunctionalHacker** [comment]
Just a "me too" comment. This would be very useful

**Comment by ShapeShifter499** [comment]
I don't know much programming as of yet, so I'm not much help here. Last time I set this up with a separate VPN there seemed to be a race-condition where OpenVPN (from F-Droid) would attempt to connect but fail out because of a WIFI captive portal and then Auto-Fi wasn't connecting because the VPN app was blocking connections till it could get through.   So if Auto-Fi happened to be first to attempt connection everything was cool but nothing worked if the VPN app was first to try a connection.
